dataplicity/limiter.py

```diff
@@ -0,0 +1,82 @@
+"""
+A threadsafe object to enforce limits.
+
+Similar in purpose to a Semaphore, but simpler because we never want to block. 
+
+"""
+from contextlib import contextmanager
+import logging
+from threading import RLock
+
+log = logging.getLogger("m2m")
+
+
+class LimiterError(Exception):
+    """Base class for counter errors."""
+
+
+class LimitReached(Exception):
+    """Count has reached the maximum"""
+
+
```

Comment:
how about adding increment and decrement as `__enter__` and `__exit__` ? then the limiter could simply be used as :

```
with Limiter( ... ):
   foo()
```